### PR TITLE
ice40: Add support for SB_I2C and SB_SPI

### PR DESCRIPTION
**Pull request by smunaut**
Signed-off-by: Sylvain Munaut <tnt@246tNt.com>

### CODE REVIEW

**Comment by janrinze** [comment]
will the merge come with examples and tests?

File: ice40/arch.cc [hunk 1]
```diff
@@ -1043,6 +1043,14 @@ TimingPortClass Arch::getPortTimingClass(const CellInfo *cell, IdString port, in
         if (port == id_CLK || port == id_CLOCK)
             return TMG_CLOCK_INPUT;
         return TMG_IGNORE;
+    } else if (cell->type == id_SB_I2C) {
+        if (port == this->id("SBCLKI"))
+            return TMG_CLOCK_INPUT;
+        return TMG_IGNORE; /* FIXME Use TMG_ENDPOINT when timing is available */
```
[the remark this answers is not in the document]
Comment (smunaut) [reply ice40/arch.cc]: @daveshah1 Just pushed an updated version

**Comment by smunaut** [comment]
@janrinze Nope.

File: ice40/arch.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (daveshah1) [reply ice40/arch.cc]: You also need to set clockInfoCount to 1, and implement getPortClockingInfo (probably by reusing the SPRAM/DSP path with a different clock port but keeping the arbitrary numbers for now) for this to be useful.

[the remark this answers is not in the document]
Comment (smunaut) [reply ice40/arch.cc]: Done. I used 1.5ns as clk-to-out and setup. When looking at the LEDDA_IP data this seemed to be the worst case.

**Comment by daveshah1** [comment]
Thanks!

**Pull request merged**